lib/ansible/modules/storage/vexata/vexata_fcinitiator.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Sandeep Kasargod (sandeep@vexata.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: vexata_fcinitiator
+version_added: 2.9
+short_description: Manage Fibre Channel initiators on Vexata VX100 storage arrays
+description:
+    - Adds or removes Fibre Channel (FC) initiators on a Vexata VX100 array.
+author:
+  - Sandeep Kasargod (@vexata)
+options:
+  name:
+    description:
+    - Initiator name.
+    required: true
+    type: str
+  state:
+    description:
+    - Add initiator when present or remove when absent.
+    - Initiators that are in one or more initiator groups cannot be deleted
+      without first deleting those initiator groups.
+    default: present
+    choices: [ present, absent ]
+    type: str
+  wwn:
+    description:
+    - FC wwn of Host HBA port for adding a new initiator.
+    type: str
+extends_documentation_fragment:
+    - vexata.vx100
+'''
+
+EXAMPLES = r'''
+- name: Add initiator named host1fc1 for host HBA port
+  vexata_initiator:
+    name: host1fc1
+    wwn: "01:23:45:67:89:ab:cd:ef"
+    state: present
+    array: vx100_ultra.test.com
+    user: admin
+    password: secret
+
+- name: Remove initiator named host1fc1
+  vexata_initiator:
+    name: host1fc1
+    state: absent
+    array: vx100_ultra.test.com
+    user: admin
+    password: secret
+'''
+
+RETURN = r'''
```

Comment:
It would be nice UI/UX by having return values of the modules.

Reply:
Currently we don't return anything. Can enhance later.